packages/jellyfish-api-core/__tests__/category/account/futureswap.test.ts

```diff
@@ -0,0 +1,744 @@
+    const tslaAddress = await testing.generateAddress()
+    await testing.rpc.account.accountToAccount(collateralAddress, { [tslaAddress]: '1@TSLA' })
+    await testing.generate(1)
+
+    // move to next settle block - 1
+    const nextSettleBlock = await getNextSettleBlock()
+    await testing.generate(nextSettleBlock - await testing.rpc.blockchain.getBlockCount() - 1)
+
+    // get minted DUSD
+    const dusdMintedBefore = (await testing.rpc.token.getToken(idDUSD))[idDUSD].minted
+
+    const swapAmount = 1
+    const fswap: FutureSwap = {
+      address: tslaAddress,
+      amount: swapAmount.toString() + '@TSLA'
+    }
+    await testing.rpc.account.futureSwap(fswap)
+    await testing.generate(1)
+
+    // check future settled
+    {
+      // calclulate minted DUSD. dtoken goes for a discount.
+      const mintedDUSD = new BigNumber((1 - futRewardPercentage) * 2 * swapAmount).dp(8, BigNumber.ROUND_FLOOR) // (1 - reward percentage) * TSLADUSD value * TSLA swap amount;
+      const dusdMintedAfter = (await testing.rpc.token.getToken(idDUSD))[idDUSD].minted
+      expect(dusdMintedAfter).toStrictEqual(dusdMintedBefore.plus(mintedDUSD))
+
+      const pendingFutures = await testing.container.call('listpendingfutureswaps')
+      expect(pendingFutures.length).toStrictEqual(0)
+
+      // check live/economy/dfip2203_*
+      const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+      expect(attributes.ATTRIBUTES['v0/live/economy/dfip2203_current']).toStrictEqual([swapAmount.toFixed(8) + '@TSLA'])
+      expect(attributes.ATTRIBUTES['v0/live/economy/dfip2203_burned']).toStrictEqual([swapAmount.toFixed(8) + '@TSLA'])
+      expect(attributes.ATTRIBUTES['v0/live/economy/dfip2203_minted']).toStrictEqual([mintedDUSD.toFixed(8) + '@DUSD'])
+
+      {
+        // check contractAddress
+        const balance = await testing.rpc.account.getAccount(contractAddress)
+        expect(balance).toStrictEqual([swapAmount.toFixed(8) + '@TSLA'])
+      }
+
+      {
+        // check tslaAddress
+        const balance = await testing.rpc.account.getAccount(tslaAddress)
+        expect(balance).toStrictEqual([mintedDUSD.toFixed(8) + '@DUSD'])
+      }
+    }
+
+    // check burn
+    const burnAfter = await testing.rpc.account.getBurnInfo()
+    expect(burnAfter.dfip2203).toStrictEqual([swapAmount.toFixed(8) + '@TSLA'])
+  })
+
+  it('should consider new oracle active price, if changed before futureswap execution', async () => {
+    const tslaAddress = await testing.generateAddress()
+    await testing.rpc.account.accountToAccount(collateralAddress, { [tslaAddress]: '1@TSLA' })
+    await testing.generate(1)
+
+    let blockHeight = await testing.rpc.blockchain.getBlockCount()
+    let nextSettleBlock = await getNextSettleBlock()
+
+    // move to next settle block for better duration for the oracle price to kick in
+    await testing.generate(nextSettleBlock - blockHeight)
+    blockHeight = await testing.rpc.blockchain.getBlockCount()
```

Comment:
You may remove this line. There is no compilation error.

Reply:
futureSwap settlement freq and oracle price update freq are two independent processes. need a sync point for the test to continue.
If you remove it, it may work for now. But in the future, let's say you add some more code and generate blocks, then the test might fail.

Reply:
Well, blockHeight is not used anywhere
<img width="571" alt="Screenshot 2022-04-07 at 6 47 19 PM" src="https://user-images.githubusercontent.com/7101806/162182420-558b5da1-521e-4b6c-abd1-8a345df47a43.png">